src/Controller/DefaultController.php

```diff
@@ -30,14 +31,31 @@ class DefaultController extends AbstractController
      */
     public function nasIndex(FundingOrgFilter $fundingOrgFilter)
     {
-        $researchGroups = array();
-
+        $filter = array();
         if ($fundingOrgFilter->isActive()) {
-            $researchGroups = $this->get('doctrine')->getRepository(ResearchGroup::class)->findBy(array('id' => $fundingOrgFilter->getResearchGroupsIdArray() ));
+            $filter = array('fundingOrganization' => $fundingOrgFilter->getFilterIdArray());
+        }
+
+        $results = $this->get('doctrine')->getRepository(FundingCycle::class)->findBy($filter, array('name' => 'ASC'));
+
+        $fundingCycles = array();
+
+        foreach ($results as $fundingCycle) {
+            $data = array();
+            $data['id'] = $fundingCycle->getId();
+            $data['name'] = $fundingCycle->getName();
+
+            foreach ($fundingCycle->getResearchGroups() as $researchGroup) {
+                $rg['id'] = $researchGroup->getId();
+                $rg['name'] = $researchGroup->getName();
+                $data['researchGroups'][] = $rg;
```

Comment:
We could use array_push

Reply:
Yes, It looks like using $array[] seems to be faster than using array_push(). Didn't know that